src/app/services/userService.tsx

```diff
@@ -0,0 +1,27 @@
+export const fetchUserProfile = async (userId: string) => {
+    const response = await fetch(`/api/users/${userId}`, {
+        method: 'GET',
+        headers: { 'Content-Type': 'application/json' },
+    });
+    if (!response.ok) throw new Error('Error fetching user profile');
+    return response.json();
+};
+
+export const updateUserProfile = async (name: string, email: string, userId: string) => {
+    const response = await fetch(`/api/users/${userId}`, {
+        method: 'PUT',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ name, email }),
+    });
+    if (!response.ok) throw new Error('Error updating profile');
+    return response.json();
+};
+
+export const fetchUsers = async() =>{
+    const response = await fetch('/api/users', {
+        method: 'GET',
+        headers: { 'Content-Type': 'application/json' },
+    });
+    if (!response.ok) throw new Error('Error fetching users');
+    return response.json();
```

Comment:
Nice. I like how you broke these functions down to a "user service". 
What are your thoughts on defining a return type for these functions to help the IntelliSense?

Reply:
That could definitely help in the future, though I'm not sure what you mean with the return type? As in define what type of information comes in the response?